changedetectionio/__init__.py

```diff
@@ -230,6 +230,30 @@ def index():
         limit_tag = request.args.get('tag')
         pause_uuid = request.args.get('pause')
 
+        if ( pause_uuid == 'resume-all' ) :
```

Comment:
What do you think about this approach instead of the kwargs?  It a ternary choice, so the logic was escaping me, but this way it validates pause_uuid if the operation is flipping a single watch's pause status, and if the operation is xxxx-all it falls through and handles it conditionally with an action var.   Just tested this and it works fine.

    def index():

        limit_tag = request.args.get('tag')
        pause_uuid = request.args.get('pause')
        
        if pause_uuid :
            try:
                # import uuid added above
                valid_uuid = UUID(pause_uuid, version=4)
                datastore.data['watching'][valid_uuid]['paused'] ^= True
            except ValueError:
        
                action = True if pause_uuid == 'pause-all' else False

                for watch_uuid, watch in datastore.data['watching'].items():
                            if datastore.data['watching'][watch_uuid]['tag'] == limit_tag or limit_tag is None :
                                datastore.data['watching'][watch_uuid]['paused'] = action

            datastore.needs_write = True
            return redirect(url_for('index', tag = limit_tag))

        ...

Reply:
This is the final version that works correctly:

    def index():

        import uuid

        limit_tag = request.args.get('tag')
        pause_uuid = request.args.get('pause')
        
        if pause_uuid :
            try:
                val = uuid.UUID(str(pause_uuid))
                datastore.data['watching'][pause_uuid]['paused'] ^= True
            except ValueError:
        
                action = True if pause_uuid == 'pause-all' else False

                for watch_uuid, watch in datastore.data['watching'].items():
                            if datastore.data['watching'][watch_uuid]['tag'] == limit_tag or limit_tag is None :
                                datastore.data['watching'][watch_uuid]['paused'] = action

            datastore.needs_write = True
            return redirect(url_for('index', tag = limit_tag))

        ...